src/renderer/components/archive/archive-search.js

```diff
@@ -0,0 +1,251 @@
+    background-color: #00b7ac;
+    color: #fff;
+    p {
+      color: #fff;
+    }
+  }
+`;
+
+const EntryData = styled.div`
+  display: inline-block;
+  margin: -5px 0 0 0;
+`;
+
+const EntryFolder = styled('p')`
+  font-size: 12px;
+  color: #999;
+  margin: 0;
+`;
+
+const Icon = styled.div`
+  float: left;
+  margin: 0 10px 0 0;
+`;
+
+class ArchiveSearch extends PureComponent {
+  static propTypes = {
+    onSelectEntry: PropTypes.func,
+    getArchive: PropTypes.func,
+    currentArchive: PropTypes.oneOfType([PropTypes.object, PropTypes.func]),
+    onGroupSelect: PropTypes.func,
+    switchArchive: PropTypes.func,
+    t: PropTypes.func
+  };
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      visible: false,
+      archive: null,
+      entries: [],
+      searchTerm: ''
+    };
+
+    this.changeInput = this.changeInput.bind(this);
+    this.closeSearch = this.closeSearch.bind(this);
+    this.searchListener = this.searchListener.bind(this);
+    this.highlightSearchResult = this.highlightSearchResult.bind(this);
+    this.getAllMatchingEntries = this.getAllMatchingEntries.bind(this);
+  }
+
+  searchListener() {
+    const { getArchive, currentArchive } = this.props;
+
+    if (currentArchive) {
+      const archive = getArchive(currentArchive.id);
+
+      this.setState(state => ({
+        visible: !state.visible,
+        archive
+      }));
+
+      this._input.focus();
+      this._input.select();
+    }
+  }
+
+  closeSearch() {
+    this.setState({
+      visible: false
+    });
+  }
+
+  changeInput(e) {
+    this.setState({
+      searchTerm: e.target.value
+    });
+
+    this.getAllMatchingEntries();
+  }
+
+  getAllMatchingEntries() {
+    getMatchingEntriesForSearchTerm(this.state.searchTerm).then(entries => {
+      this.setState({
+        entries: this.state.searchTerm ? entries : []
+      });
+    });
+  }
+
+  highlightSearchResult(word) {
+    const regex = new RegExp('(' + this.state.searchTerm + ')', 'g');
+    return word.replace(regex, '<mark>$1</mark>');
+  }
+
+  componentDidMount() {
+    ipcRenderer.on('open-archive-search', this.searchListener);
+  }
+
+  componentWillUnmount() {
+    ipcRenderer.removeListener('open-archive-search', this.searchListener);
+  }
+
+  render() {
+    const { entries, searchTerm } = this.state;
+    const { switchArchive, onSelectEntry, onGroupSelect, t } = this.props;
+
+    return (
+      <SearchWrapper visible={this.state.visible}>
+        <SearchOverlay onClick={this.closeSearch} />
+        <Search flexColumn>
+          <Input
+            bordered
+            innerRef={input => {
+              this._input = input;
+            }}
+            onChange={this.changeInput}
+            value={this.state.searchTerm}
+            placeholder={t('archive-search.searchterm')}
+            type="text"
+            entries={this.state.entries}
+            searchTerm={this.state.searchTerm}
+          />
+
+          {entries.length > 0 ? (
```

Comment:
Instead of this ternary operator, you can use `<Choose>` and `<When>`, we have been using them in other files
https://www.npmjs.com/package/jsx-control-statements